src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -118,18 +115,72 @@ public boolean isRunning() {
         return scheduledFuture != null;
     }
 
+    /**
+     * Returns the next Instant the task is scheduled to run. This will change based on whether the device goes offline
+     * or not but the guarantee is that it is scheduled to run 24h after the last successful run.
+     */
+    public Instant getNextScheduledRun() {
+       return nextScheduledRun.get();
+    }
+
+    /**
+     * Returns the last Instant the task ran successfully.
+     */
+    public Instant getLastRan() {
+        return lastRan.get();
+    }
```

Comment:
Not a huge fan of these methods - right now you're just using these in tests. That's ok, but I'd prefer to test actual observable behavior instead of asking the class what it did / intends to do.

Reply:
Agree... the tricky part was testing the scheduler. Technically I could make it run every second but the non determinism would make the test flaky this way is not perfect but gave me a sense of whether it was doing the right thing or not?. Do you know of a better way we could test it?